tpcp/_dataset.py

```diff
@@ -64,6 +65,10 @@ def _create_check_index(self):
         While we can not catch all related issues (i.e. determinism across different machines), this should catch the
         most obvious ones.
 
+        Furthermore, we check if all columns of the index are valid Python attribute names and throw a warning if not.
+        A string is a valid Python attribute name, if it only contains alphanumeric letters, or underscores,
```

Comment:
Can you move this explanaation into the warning? The user is unlikely to read the docsting here, before running into this issue